src/components/progress/progress.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, Input, ViewEncapsulation, HostBinding } from '@angular/core';
+
+@Component({
+  selector: 'ion-progress',
+  templateUrl : './progress.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class Progress {
+
+  constructor( ) { }
+
+  /**
+   * @input {'determinate' | 'indeterminate' | 'buffer' | 'query'} The current mode of the progress indicator
+   */
+  @Input()
+  @HostBinding('class')
```

Comment:
instead of using `HostBinding`, usually they use `host` on `Component`, see [range](https://github.com/ionic-team/ionic/blob/master/src/components/range/range.ts#L99)